android/app/src/main/java/com/microsoft/codepush/react/CodePush.java

```diff
@@ -439,19 +441,43 @@ public void run() {
         }
 
         @ReactMethod
-        public void downloadUpdate(final ReadableMap updatePackage, final Promise promise) {
+        public void downloadUpdate(final ReadableMap updatePackage, final boolean notifyProgress, final Promise promise) {
             AsyncTask<Void, Void, Void> asyncTask = new AsyncTask<Void, Void, Void>() {
                 @Override
                 protected Void doInBackground(Void... params) {
                     try {
                         WritableMap mutableUpdatePackage = CodePushUtils.convertReadableMapToWritableMap(updatePackage);
                         mutableUpdatePackage.putString(BINARY_MODIFIED_TIME_KEY, "" + getBinaryResourcesModifiedTime());
                         codePushPackage.downloadPackage(mutableUpdatePackage, CodePush.this.assetsBundleFileName, new DownloadProgressCallback() {
+                            private boolean nextFrameBusy = false;
```

Comment:
How about calling this `hasScheduledNextFrame`, or `isNextFramePending` or something similar? "Busy" seems a little odd in this context, so we're really talking about an event that we don't need to schedule multiple times, as opposed to an operation that can take too long.